### ISSUE TITLE
setup dynamic remote in v5  federation v2

### ISSUE THREAD

**Issue by hosseinmd** [callstack/repack#815]
### Describe the bug

i had a federation config like 
`
remotes: {
     app1:"app1@dynamic"
}
`
and I set up a resolver to handle chunks address dynamically. Now it's not working with federation v2

### System Info

```shell
Repack webpack federationV2
```

### Re.Pack Version

5.0.0-rc.2

### Reproduction

.

### Steps to reproduce

.

**Comment by hosseinmd** [comment]
<img src="https://github.com/user-attachments/assets/829b760a-356d-4c0a-a459-ec08dc939bdf" width="500"/>

When I use loadRemote this error happened

**Comment by jbroma** [comment]
this shouldn't be happening, can you provide a simple repro with this error?

**Comment by hosseinmd** [comment]
https://github.com/callstack/super-app-showcase/
it is working by rspack but not working by webpack.

**Comment by jbroma** [comment]
@hosseinmd yeah, I remember having some issues with webpack, for now please use Rspack to achieve that.

**Comment by hosseinmd** [comment]
Probably it is because getInitializationEntries is removed, have you any replacement?

**Comment by jbroma** [comment]
> Probably it is because getInitializationEntries is removed, have you any replacement?

`getInitializationEntries` was removed for now because it is now handled internally and should work out of the box, I'll investigate and revert this change if needed for the stable release

**Comment by hosseinmd** [comment]
Referring to https://module-federation.io/guide/basic/runtime.html#registerremotes 
for dynamic remote we should use
```import { init, preloadRemote } from '@module-federation/enhanced/runtime';```

i added it to my own project and also https://github.com/callstack/super-app-showcase/ i got this error

```
Module not found: Can't resolve '@module-federation/enhanced/runtime' in '/Users/hossein/Documents/github/super-app-showcase/packages/host'
   ╭─[6:21]
 4 │ import { name as appName } from './app.json';
 5 │ import AsyncStorage from '@react-native-async-storage/async-storage';
 6 │ import { init } from '@module-federation/enhanced/runtime';
   ·                      ─────────────────────────────────────
 7 │ init({
 8 │     name: 'host',
```

**Comment by hosseinmd** [comment]
i changed it to 
```@module-federation/runtime``` and know i have another error
<img width="363" alt="Image" src="https://github.com/user-attachments/assets/a7d4d442-cb5a-49c4-859d-d653700c115a" />

**Comment by hosseinmd** [comment]
this will fix it

```
import { init } from '@module-federation/runtime';
```

**Comment by jbroma** [comment]
> this will fix it
> 
> ```
> import { init } from '@module-federation/runtime';
> ```

glad you figured this one out, any other issues with the dynamic remotes?

**Comment by hosseinmd** [comment]
Hi @jbroma thanks for followup.
i have no a problem in develop mode which other containers do not know what remotes are initialized in main container for solving this i add a patch to @module-federation/runtime-core
```js
-            remotes: [],
+            remotes: CurrentGlobal.__FEDERATION__?.remotes || [],
             shared: {},
```
dev mode working know

but on release mode i have crash, i don't know why yet and i will post it here if i found what is problem

**Comment by jbroma** [comment]
> Hi [<img alt="" width="16" height="16" src="https://avatars.githubusercontent.com/u/20626315?u=8cacaea406d594b6428c4c22a75332996d7a5c87&amp;v=4&amp;size=80">@jbroma](https://github.com/jbroma) thanks for followup. i have a problem in develop mode which other containers do not know what remotes are initialized in main container for solving this i add a patch to @module-federation/runtime-core
> 
> -            remotes: [],
> +            remotes: CurrentGlobal.__FEDERATION__?.remotes || [],
>              shared: {},
> dev mode working know
> 
> but on release mode i have crash, i don't know why yet and i will post it here if i found what is problem

please open a new issue for that and we can investigate this problem 👍
